src/DotNet.Status.Web/Controllers/AnnotationsController.cs

```diff
@@ -0,0 +1,109 @@
+using Azure.Data.Tables;
+using DotNet.Status.Web.Models;
+using DotNet.Status.Web.Options;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.DotNet.Services.Utility;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace DotNet.Status.Web.Controllers
+{
+    /// <summary>
+    /// This Annotations controller serves queries in a format compatible with the Grafana
+    /// "Simple JSON Datasource". It is used to expose the "Deployments" table data to 
+    /// Grafana, which may then render the information in dashboards.
+    /// </summary>
+    [Route("api/[controller]")]
+    [ApiController]
+    public class AnnotationsController : ControllerBase
+    {
+        private const int _maximumServerCount = 10; // Safety limit on query complexity
+        private readonly ILogger<AnnotationsController> _logger;
+        private readonly IOptionsMonitor<AnnotationsOptions> _options;
+
+        public AnnotationsController(ILogger<AnnotationsController> logger, IOptionsMonitor<AnnotationsOptions> options)
+        {
+            _logger = logger;
+            _options = options;
+        }
+
+        [HttpGet]
+        public ActionResult Get()
+        {
+            return Ok();
+        }
+
+        [HttpPost]
+        [Route("query")]
+        public ActionResult Query()
+        {   // Required endpoint, but not used
+            return NoContent();
+        }
+
+        [HttpPost]
+        [Route("search")]
+        public ActionResult Search()
+        {   // Required endpoint, but not used
+            return NoContent();
+        }
+
+        [HttpPost]
+        [Route("annotations")]
+        public async IAsyncEnumerable<AnnotationEntry> Post(AnnotationQueryBody annotationQuery, [EnumeratorCancellation] CancellationToken cancellationToken)
+        {
+            TableClient tableClient = new TableClient(new Uri(_options.CurrentValue.DeploymentsTableConnectionString));
+
+            IEnumerable<string> services = (annotationQuery.Annotation.Query?.Split(',') ?? Array.Empty<string>())
+               .Take(_maximumServerCount)
```

Comment:
There is no sort of ordering mentioned here... making this "Task" a little random?  We just... ignore some of the requests, I guess?

Reply:
Do you mean "Take" instead of "Task"? I added it only as a cheap way to sanitize external query input. It shouldn't happen in normal use (we have fewer services than the limit, and if that's not true then the limit should probably just be raised).

Reply:
I did. I did mean Take.  Hrm... we could return a "400 bad request" here instead?  That's a little easier to track down than randomly losing some set of input with no indication why. :-) At least we'd be able to see that in the logs and address it.

Reply:
I would much prefer an error here if we receive more than the maximum.